app/models/group.rb

```diff
@@ -414,6 +414,16 @@ def bulk_add(user_ids)
       if self.title.present?
         User.where(id: user_ids).update_all(title: self.title)
       end
+
+      if self.grant_trust_level.present?
+        User.where(id: user_ids).find_each do |user|
```

Comment:
I looked into that approach (raw SQL) but it seems like `user.change_trust_level!` calls `Promotion.new(self).change_trust_level!` which is [doing many things](https://github.com/discourse/discourse/blob/5dbd6a304bed5400be481d71061d3e3ebb4d6785/lib/promotion.rb#L39-L78), so it might not be possible to accommodate all that in raw SQL?

The same concern applies for using `update_all` directly.